client/blocks/site-preview/index.jsx

```diff
@@ -61,22 +40,33 @@ function SitePreview( { className } ) {
 				'frame-nonce': selectedSiteNonce ?? '',
 				cachebust: previewCount,
 			},
-			basePreviewUrl
+			previewUrl
 		);
 	}
 
 	return (
 		<WebPreview
 			className={ className }
 			previewUrl={ formatPreviewUrl() }
-			externalUrl={ basePreviewUrl }
+			externalUrl={ previewUrl }
 			showExternal={ true }
 			showClose={ true }
 			showPreview={ showPreview }
-			onClose={ () => dispatch( closePreview() ) }
+			onClose={ () => {
+				dispatch( closePreview() );
+				incrementPreviewCount();
+			} }
 			showSEO={ ! hideSEO }
 		/>
 	);
 }
 
-export default SitePreview;
+const withSiteIdAsKey = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const selectedSiteId = useSelector( getPreviewSiteId );
+		return <Wrapped key={ `site-preview-${ selectedSiteId }` } { ...props } />;
+	},
+	'WithSiteIdAsKey'
+);
+
+export default withSiteIdAsKey( SitePreview );
```

Comment:
style nit: I'd personally wouldn't bother with creating a HOC and would write it directly:
```js
export default function SitePreview( props ) {
  const siteId = useSelector( getPreviewSiteId );
  return <SitePreviewInner key={ siteId } siteId={ siteId } { ...props } />;
}
```
My example also omits the `site-preview-` prefix from the key (it's not part of any namespace where it would need to be globally unique) and passes the `siteId` prop down so that it doesn't need to be selected again.

Reply:
Agreed. Reworked in https://github.com/Automattic/wp-calypso/pull/63008/commits/a3daf590f750bc3876f87328889ea29a217302be.